activesupport/lib/active_support/core_ext/numeric/conversions.rb

```diff
@@ -114,32 +117,16 @@ def to_formatted_s(format = :default, options = {})
     when :human_size
       return ActiveSupport::NumberHelper.number_to_human_size(self, options)
     else
-      self.to_default_s
-    end
-  end
-
-  [Fixnum, Bignum].each do |klass|
-    klass.class_eval do
-      alias_method :to_default_s, :to_s
-      def to_s(base_or_format = 10, options = nil)
-        if base_or_format.is_a?(Symbol)
-          to_formatted_s(base_or_format, options || {})
-        else
-          to_default_s(base_or_format)
-        end
-      end
+      super
     end
   end
 
-  Float.class_eval do
-    alias_method :to_default_s, :to_s
-    def to_s(*args)
-      if args.empty?
-        to_default_s
-      else
-        to_formatted_s(*args)
-      end
-    end
+  def to_formatted_s(*args)
+    to_s(*args)
   end
+  deprecate to_formatted_s: :to_s
+end
 
+[Fixnum, Bignum, Float, BigDecimal].each do |klass|
```

Comment:
I would be delighted to do so, but we've never overwritten `Complex` and `Rational` so I hesitated a little:

```ruby
>> ObjectSpace.each_object(Class).select { |klass| klass < Numeric }
#=> [Complex, Rational, Bignum, Float, Fixnum, Integer]
```
and it looks like it will not work properly:
```ruby
>> ActiveSupport::NumberHelper.number_to_percentage Complex(2,2)
#=> RangeError: can't convert 2+2i into Float
```